### PR TITLE
Dh preflight checks

### LINKED ISSUE #96: Checking for new Cql updates before locks are acquired
In the SAS Vali team we have been experiencing issues with acquiring locks and releasing locks and were therefore wondering if it's possible to make the check to see if there are Cql changes to migrate before the locks are acquired.

### PR DESCRIPTION
**Pull request by SteveWink**
This pull request is to provide functionality to check whether a cql migrate should be carried out or not, before taking up resources such as locks.
It checks that the keyspace exists. If not it will migrate.
It checks that the table exists. If not it will migrate.
It checks whether the files specified have the same checksum as the files already in the database with the same names. If any of the files are missing  it will migrate.

If a row exists for a file but it has a different checksum an exception will be thrown, as before but with a clearer message.

Fixes #96

### CODE REVIEW

File: src/test/java/uk/sky/cqlmigrate/PreMigrationCheckerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    @Test
+    public void migrationNotNeededIfEverythingExists() throws Exception {
+        //given
+        cluster.connect("system").execute("CREATE KEYSPACE " + TEST_KEYSPACE + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1 };");
+        cluster.connect(TEST_KEYSPACE).execute("CREATE TABLE schema_updates (filename text primary key, checksum text, applied_on timestamp);");
+
+        cluster.connect(TEST_KEYSPACE).execute("INSERT INTO schema_updates (filename, checksum, applied_on) VALUES ('2015-04-01-13:56-create-status-table.cql', 'fa03a30eab18b64b74ee1ea7816e0513f03b4ac7', dateof(now()));");
+        cluster.connect(TEST_KEYSPACE).execute("INSERT INTO schema_updates (filename, checksum, applied_on) VALUES ('2015-04-01-13:57-add-column-to-status-table.cql', '93abe7ec3b14888b9a8bd8d680a30839a92a3248', dateof(now()));");
```
Comment (chuckydev) [inline src/test/java/uk/sky/cqlmigrate/PreMigrationCheckerIntegrationTest.java:63]: These commands are duplicated. Could we share them in some way to make this easier to test in the future? (Both 
 lines 62/63 seem duplicated)

**Review by chuckydev — CHANGES_REQUESTED** [review]
Thanks for the contribution all. Just a nice to have in test springs to mind. There is a little bit of duplication of creations that may be hard to understand in the future. Could these be factored into something more descriptive?

File: src/test/java/uk/sky/cqlmigrate/PreMigrationCheckerIntegrationTest.java (hunk 1 above)
Comment (davidh87) [reply src/test/java/uk/sky/cqlmigrate/PreMigrationCheckerIntegrationTest.java]: Attempted cleanup!

**Comment by davidh87** [comment]
@jonathanpeircesky @fincefid As people that previously expressed interest; does this look like it addresses #96 enough from your POVs?

**Review by chuckydev — APPROVED** [review]

**Pull request merged**